src/V3AstNodes.h

```diff
@@ -4835,27 +4836,33 @@ class AstTraceDecl : public AstNodeStmt {
     AstBasicDTypeKwd declKwd() const { return m_declKwd; }
     VDirection declDirection() const { return m_declDirection; }
     bool isScoped() const { return m_isScoped; }
+    AstNode* valuep() const { return op1p(); }
 };
 
 class AstTraceInc : public AstNodeStmt {
-    // Trace point; incremental change detect and dump
+    // Trace point dump
     // Parents:  {statement list}
-    // Children: incremental value
+    // Children: op1: things to emit before this node,
+    //           op2: expression being traced (from decl)
+
 private:
-    AstTraceDecl* m_declp;  // [After V3Trace] Pointer to declaration
+    AstTraceDecl* m_declp;  // Pointer to declaration
+    const bool m_full;  // Is this a full vs incremental dump
 public:
-    AstTraceInc(FileLine* fl, AstTraceDecl* declp, AstNode* valuep)
-        : ASTGEN_SUPER(fl) {
+    AstTraceInc(FileLine* fl, AstTraceDecl* declp, bool full)
+        : ASTGEN_SUPER(fl)
+        , m_declp(declp)
+        , m_full(full) {
         dtypeFrom(declp);
-        m_declp = declp;
-        addNOp2p(valuep);
+        addOp2p(declp->valuep()->cloneTree(true));
     }
     ASTNODE_NODE_FUNCS(TraceInc)
     virtual const char* broken() const {
         BROKEN_RTN(!declp()->brokeExists());
         return NULL;
     }
     virtual void cloneRelink() {
+        // Is this ever needed?
```

Comment:
```suggestion
```

Might not be called, but as policy want these always in place for any tree object with pointer.